File: net/merge/Merger_bmp_pdf.cs
namespace aspose_snippets.net
{
    public static partial class Merger
    {
        public static void bmp_to_pdf()
        {
            const string pathSource1 = "../../TestData/test.bmp";
            const string pathSource2 = "../../TestData/Second/test.bmp";

            //create empty pdf document
            using var doc = new Aspose.Pdf.Document();

            Aspose.Pdf.Rectangle rect;

            //make list of file streams with documents to merge with documents to merge
            var streams = new List<Stream>()
            {
                File.Open(pathSource1, FileMode.Open),
                File.Open(pathSource2, FileMode.Open)
            };

            foreach (var fs in streams)
            {
                //add new page to pdf
                Aspose.Pdf.Page page = doc.Pages.Add();

                //setup page size to be A4
                page.SetPageSize(
                    Aspose.Pdf.PageSize.A4.Width,
                    Aspose.Pdf.PageSize.A4.Height);

                //load image from stream, it supports a lot of formats
                using (Aspose.Imaging.Image image = Aspose.Imaging.Image.Load(fs))
                {
                    //read image dimensions to pdf page rectangle
                    rect = new Aspose.Pdf.Rectangle(0, 0, image.Width - 1, image.Height - 1);
                }

                //reset stream to read from begin for next step
                fs.Seek(0, SeekOrigin.Begin);

                //add image to new pdf page
                page.AddImage(fs, rect);
            }

            //save result pdf to file
            doc.Save("Merger_bmp_pdf.pdf", Aspose.Pdf.SaveFormat.Pdf);
        }
    }
}

File: java/merge/src/aspose_snippets/java/Merger_bmp_pdf.java
package aspose_snippets.java;

import java.io.FileNotFoundException;
import java.util.Arrays;
import java.util.List;

public class Merger_bmp_pdf {
    public static void Execute() throws FileNotFoundException {
        String pathSource1 = "../../../../TestData/test.bmp";
        String pathSource2 = "../../../../TestData/Second/test.bmp";

        //create empty pdf document
        com.aspose.pdf.Document doc = new com.aspose.pdf.Document();

        //make list of files with images to merge
        List<String> images = Arrays.asList(new String[]{ pathSource1, pathSource2 });

        for(String fs : images)
        {
            //add new page to pdf
            com.aspose.pdf.Page page = doc.getPages().add();

            //setup page size to be A4
            page.setPageSize(
                    com.aspose.pdf.PageSize.getA4().getWidth(),
                    com.aspose.pdf.PageSize.getA4().getHeight());

            com.aspose.pdf.Rectangle rect;

            //load image from stream, it supports a lot of formats
            com.aspose.imaging.Image image = com.aspose.imaging.Image.load(fs);
            //read image dimensions to pdf page rectangle
            rect = new com.aspose.pdf.Rectangle(0, 0, image.getWidth() - 1, image.getHeight() - 1);

            //add image to new pdf page
            page.addImage(fs, rect);
        }

        //save result pdf to file
        doc.save("test.pdf", com.aspose.pdf.SaveFormat.Pdf);
    }
}
